templateData.go

```diff
@@ -98,14 +98,16 @@ type MetricsData struct {
 
 // SidecarData configures the injected sidecar
 type SidecarData struct {
-	UseOpenrestySidecar  bool
-	Image                string
-	HealthCheckPath      string
-	EnvironmentVariables map[string]interface{}
-	CPURequest           string
-	MemoryRequest        string
-	CPULimit             string
-	MemoryLimit          string
+	Type                     string
+	Image                    string
+	HealthCheckPath          string
+	DbInstanceConnectionName string
+	SQLProxyPort             string
```

Comment:
I'm not sure how this should be handled as more sidecar types are introduced, because then more and more specific fields will be mixed together here. Maybe we can extract the specific ones into small specific structs if this becomes unmaintainable.